lib/all_changed_attributes.rb

```diff
@@ -0,0 +1,47 @@
+require 'all_changed_attributes/version'
+
+module AllChangedAttributes
+  def write_attribute(attr_name, value)
+    store_all_changed_attributes(attr_name, value) if attributes[attr_name] != value
+    super
+  end
+
+  def all_changed_attributes(new = {})
+    new.merge(@all_changed_attributes ||= {})
+  end
+
+  def store_all_changed_attributes(attr_name, value)
```

Comment:
The `value` parameter isn't used.  Instead you are using `attributes[attr_name]`